Include/internal/pycore_unionobject.h

```diff
@@ -8,13 +8,13 @@ extern "C" {
 #  error "this header requires Py_BUILD_CORE define"
 #endif
 
-PyAPI_DATA(PyTypeObject) _PyUnion_Type;
```

Comment:
All the ``extern`` keywords here are optional right?

Reply:
It is required for variable declarations and optional for function declarations. But other function declarations in headers have `extern`.

Reply:
I see, thanks. This LGTM. But you might have trouble backporting to 3.10 due to the ABI check.

Reply:
Actually the ABI check does not complain when we stop exporting some symbols. It only complains if we add some new exported symbols.

Reply:
> Actually the ABI check does not complain when we stop exporting some symbols. It only complains if we add some new exported symbols.

Fixing this in https://github.com/python/cpython/pull/27229. The check is backwards